File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    stk,i=[],0
    while i<len(arr):
        if not len(stk):
            stk+=[arr[i]]
        else:
            if stk[-1]==arr[i]:
                stk.pop()
            else:
                stk+=[arr[i]]
        i+=1
    return [-1] if not len(stk) else stk

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.cpp
#include <string>
#include <vector>
using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> r;
    for(int i=0;i<arr.size();i++) {
        if(r.size()==0)
            r.push_back(arr[i]);
        else {
            if(r[r.size()-1]==arr[i])
                r.pop_back();
            else
                r.push_back(arr[i]);
        }
    }
    if(r.size()==0)
        r.push_back(-1);
    return r;
}